cedar-policy/src/api.rs

```diff
@@ -3924,3 +3924,91 @@ mod partial_eval_test {
         assert_eq!(a.residuals(), &p);
     }
 }
+
+#[cfg(test)]
+// PANIC SAFETY: unit tests
+#[allow(clippy::unwrap_used)]
+mod test {
+    use super::*;
+
+    #[test]
+    fn test_all_ints() {
+        test_single_int(0);
+        test_single_int(i64::MAX);
+        test_single_int(i64::MIN);
+    }
+
+    fn test_single_int(x: i64) {
+        for i in 0..4 {
+            test_single_int_with_dashes(x, i);
+        }
+    }
+
+    fn test_single_int_with_dashes(x: i64, dashes: usize) {
+        let dashes = vec!['-'; dashes].into_iter().collect::<String>();
+        let src = format!(r#"permit(principal, action, resource) when {{ {dashes}{x} }};"#);
+        let p: Policy = src.parse().unwrap();
+        let json = p.to_json().unwrap();
+        let round_trip = Policy::from_json(None, json).unwrap();
+        let pretty_print = format!("{round_trip}");
+        assert!(pretty_print.contains(&x.to_string()));
+    }
+
+    // Serializing a valid 64-bit int that can't be represented in double precision float
+    #[test]
```

Comment:
A little confused about this comment: why is converting this int to json ok?

Reply:
From the JSON spec
> JSON is agnostic about the semantics of numbers. In any programming language, there can be a variety of
number types of various capacities and complements, fixed or floating, binary or decimal. That can make
interchange between different programming languages difficult. JSON instead offers only the representation of
numbers that humans use: a sequence of digits. All programming languages know how to make sense of digit
sequences even if they disagree on internal representations. That is enough to allow interchange

Serde in particular is happy to either ser/de JSON numbers as floats or ints, depending on what datastructure you're ser/deing to/from. This does mean that applications interfacing from Cedar whose JSON libraries take a different approach could lead to loss of precision.